From 11130b14371f9ee0de6a203802a6499eed5b6571 Mon Sep 17 00:00:00 2001
From: Alexander Rukletsov <alexr@apache.org>
Date: Tue, 21 Aug 2018 17:59:27 +0200
Subject: [PATCH] Revert "Removed long ago deprecated endpoints.".

This reverts commit 42551cb5290b7b04101f7d800b4b8fd573e47b91.

Even though the endpoints have been deprecated long ago and should
have already been removed, doing this so close to the release might
prevent people from adjusting their tooling on time.

The deprecated endpoints are restored in 1.7.x only, Mesos master
aka 1.8.0-dev will not have deprecated endpoints.

Signed-off-by: Gilbert Song <songzihao1990@gmail.com>
---
 src/files/files.cpp   | 19 +++++++++++++++++++
 src/master/master.cpp | 33 +++++++++++++++++++++++++++++++++
 src/slave/slave.cpp   | 21 +++++++++++++++++++++
 3 files changed, 73 insertions(+)

diff --git a/src/files/files.cpp b/src/files/files.cpp
index f200d5e..a64613c 100644
--- a/src/files/files.cpp
+++ b/src/files/files.cpp
@@ -222,6 +222,25 @@ FilesProcess::FilesProcess(
 
 void FilesProcess::initialize()
 {
+    // TODO(ijimenez): Remove these endpoints at the end of the
+    // deprecation cycle on 0.26.
+    route("/browse.json",
+          authenticationRealm,
+          FilesProcess::BROWSE_HELP,
+          &FilesProcess::loggedBrowse);
+    route("/read.json",
+          authenticationRealm,
+          FilesProcess::READ_HELP,
+          &FilesProcess::loggedRead);
+    route("/download.json",
+          authenticationRealm,
+          FilesProcess::DOWNLOAD_HELP,
+          &FilesProcess::loggedDownload);
+    route("/debug.json",
+          authenticationRealm,
+          FilesProcess::DEBUG_HELP,
+          &FilesProcess::loggedDebug);
+
     route("/browse",
           authenticationRealm,
           FilesProcess::BROWSE_HELP,
diff --git a/src/master/master.cpp b/src/master/master.cpp
index fbde112..60487be 100644
--- a/src/master/master.cpp
+++ b/src/master/master.cpp
@@ -966,6 +966,16 @@ void Master::initialize()
           logRequest(request);
           return http.reserve(request, principal);
         });
+  // TODO(ijimenez): Remove this endpoint at the end of the
+  // deprecation cycle on 0.26.
+  route("/roles.json",
+        READONLY_HTTP_AUTHENTICATION_REALM,
+        Http::ROLES_HELP(),
+        [this](const process::http::Request& request,
+               const Option<Principal>& principal) {
+          logRequest(request);
+          return http.roles(request, principal);
+        });
   route("/roles",
         READONLY_HTTP_AUTHENTICATION_REALM,
         Http::ROLES_HELP(),
@@ -993,6 +1003,19 @@ void Master::initialize()
               logResponse(request, response);
             });
         });
+  // TODO(ijimenez): Remove this endpoint at the end of the
+  // deprecation cycle on 0.26.
+  route("/state.json",
+        READONLY_HTTP_AUTHENTICATION_REALM,
+        Http::STATE_HELP(),
+        [this](const process::http::Request& request,
+               const Option<Principal>& principal) {
+          logRequest(request);
+          return http.state(request, principal)
+            .onReady([request](const process::http::Response& response) {
+              logResponse(request, response);
+            });
+        });
   route("/state",
         READONLY_HTTP_AUTHENTICATION_REALM,
         Http::STATE_HELP(),
@@ -1015,6 +1038,16 @@ void Master::initialize()
               logResponse(request, response);
             });
         });
+  // TODO(ijimenez): Remove this endpoint at the end of the
+  // deprecation cycle.
+  route("/tasks.json",
+        READONLY_HTTP_AUTHENTICATION_REALM,
+        Http::TASKS_HELP(),
+        [this](const process::http::Request& request,
+               const Option<Principal>& principal) {
+          logRequest(request);
+          return http.tasks(request, principal);
+        });
   route("/tasks",
         READONLY_HTTP_AUTHENTICATION_REALM,
         Http::TASKS_HELP(),
diff --git a/src/slave/slave.cpp b/src/slave/slave.cpp
index 9c14784..426c389 100644
--- a/src/slave/slave.cpp
+++ b/src/slave/slave.cpp
@@ -795,6 +795,17 @@ void Slave::initialize()
 
         return resourceProviderManager->api(request, principal);
       });
+
+  // TODO(ijimenez): Remove this endpoint at the end of the
+  // deprecation cycle on 0.26.
+  route("/state.json",
+        READONLY_HTTP_AUTHENTICATION_REALM,
+        Http::STATE_HELP(),
+        [this](const http::Request& request,
+               const Option<Principal>& principal) {
+          logRequest(request);
+          return http.state(request, principal);
+        });
   route("/state",
         READONLY_HTTP_AUTHENTICATION_REALM,
         Http::STATE_HELP(),
@@ -827,6 +838,16 @@ void Slave::initialize()
           logRequest(request);
           return http.statistics(request, principal);
         });
+  // TODO(ijimenez): Remove this endpoint at the end of the
+  // deprecation cycle on 0.26.
+  route("/monitor/statistics.json",
+        READONLY_HTTP_AUTHENTICATION_REALM,
+        Http::STATISTICS_HELP(),
+        [this](const http::Request& request,
+               const Option<Principal>& principal) {
+          logRequest(request);
+          return http.statistics(request, principal);
+        });
   route("/containers",
         READONLY_HTTP_AUTHENTICATION_REALM,
         Http::CONTAINERS_HELP(),
-- 
2.5.1

